pkg/reconciler/knativeserving/controller.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019 The Knative Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package knativeserving
+
+import (
+	"context"
+	"flag"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"os"
+	"path/filepath"
+
+	mf "github.com/jcrossley3/manifestival"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	deploymentinformer "knative.dev/pkg/injection/informers/kubeinformers/appsv1/deployment"
+	serviceinformer "knative.dev/pkg/injection/informers/kubeinformers/corev1/service"
+	"knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+	knativeServinginformer "knative.dev/serving-operator/pkg/client/injection/informers/serving/v1alpha1/knativeserving"
+	rbase "knative.dev/serving-operator/pkg/reconciler/newreconciler"
+)
+
+const (
+	controllerAgentName = "knativeserving-controller"
+	reconcilerName = "KnativeServing"
+)
+
+var (
+	ClusterConfig *rest.Config
```

Comment:
We shouldn't use globals which we update at runtime, especially not in pkg. Can we not pass this value into the controller itself? Or just refetch it inside of the controller if that's not easily possible.